GliaWidgets/Sources/ViewModel/Call/CallViewModel.swift

```diff
@@ -301,6 +271,36 @@ class CallViewModel: EngagementViewModel, ViewModel {
 }
 
 extension CallViewModel {
+    private func handleAudioStreamError(_ error: CoreSdkClient.SalemoveError) {
+        switch error.error {
+        case let mediaError as CoreSdkClient.MediaError:
+            switch mediaError {
+            case .permissionDenied:
+                showSettingsAlert(
+                    with: alertConfiguration.microphoneSettings
+                )
+            default:
+                showAlert(for: error)
+            }
+        default:
+            showAlert(for: error)
+        }
```

Comment:
would it make sense to use the `where` keyword to simplify the statement?
```swift
        switch error.error {
        case let mediaError as CoreSdkClient.MediaError where mediaError == .permissionDenied:
            showSettingsAlert(
                with: alertConfiguration.microphoneSettings
            )
        default:
            showAlert(for: error)
        }
```

Reply:
perhaps we even can union two function to avoid boilerplate and have more universal function like.
```swift
    private func handleStreamError(
        _ error: CoreSdkClient.SalemoveError,
        with message: SettingsAlertConfiguration
    ) {
        switch error.error {
        case let mediaError as CoreSdkClient.MediaError where mediaError == .permissionDenied:
            showSettingsAlert(
                with: message
            )
        default:
            showAlert(
                for: error
            )
        }
    }
```
what do you think?